By Michael Holt, as part of team 3 for the DataPhilly 2022 Datathon to help ElderNet.

In [1]:
import numpy as np
import pandas as pd

# Importing Data

In [2]:
donations_df = pd.read_csv('data/donations_anonymized.csv')

# Largest Donation Amount

In [3]:
donations_df.head(10)

anon_donor_id      zip  status  do_not_mail  do_not_call organisation  \
0             92  19003.0  Active        False        False            N   
1            321  19096.0  Active        False        False            N   
2            559  19146.0  Active        False        False            N   
3            926  19041.0  Active        False        False            N   
4            860  10038.0  Active        False        False            N   
5            956  19003.0  Active        False        False            N   
6            835  19072.0  Active        False        False            N   
7            119  19041.0  Active        False        False            N   
8            485  19041.0  Active        False        False            N   
9            192  19403.0  Active        False        False            N   

         date  amount   form    campaign target  
0  01/01/2019    40.0  Check     Clients   Gift  
1  01/02/2019    25.0  Check  Fall Towns   Gift  
2  01/02/2019   250.0  Check  Fall Towns   Gift  
3  01/03/2019    50.0  Check  Fall Towns   Gift  
4  01/04/2019   165.0  Check        Misc   Gift  
5  01/07/2019    32.0  Check  Fall Towns   Gift  
6  01/07/2019   100.0  Check       Board   Gift  
7  01/08/2019   100.0  Check       Board   Gift  
8  01/09/2019    50.0  Check  Fall Towns   Gift  
9  01/09/2019  1000.0  Check  Corporatio   Gift

The largest single donation recorded here can be found through max().

In [4]:
max(donations_df.amount)

186092.0

A filter for all rows which have this value. This will help add some context.

In [5]:
donations_df.loc[: , "amount"] == max(donations_df.amount)

0       False
1       False
2       False
3       False
4       False
        ...  
2185    False
2186    False
2187    False
2188    False
2189    False
Name: amount, Length: 2190, dtype: bool

In [6]:
mask = donations_df.loc[: , "amount"] == max(donations_df.amount)

In [7]:
donations_df[mask]

anon_donor_id  zip  status  do_not_mail  do_not_call organisation  \
1809            589  NaN  Active        False        False            N   

            date    amount   form campaign target  
1809  01/14/2021  186092.0  Check     Misc   Gift

This above row has data on the largest recorded donation. There was no tie for the biggest donation. The biggest donation occurred on 1/14/2021, was not from an organization, and was a check for a whopping $186,092.